cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -414,7 +414,12 @@ where
 
     /// Return the name of rule `ridx`. Panics if `ridx` doesn't exist.
     pub fn rule_name(&self, ridx: RIdx<StorageT>) -> &str {
```

Comment:
For consistency (and clarity) I reckon this should be either:

```rust
pub fn rule_name(&self, ridx: RIdx<StorageT>) -> (&str, Span)
```

or the two functions are `rule_name_str` and `rule_name_span`.

Reply:
Sounds okay to me, do we just want that one `rule_name (..) -> (&str, Span)` or all three?
I had mostly left it how it was before strictly for backwards compatibility, but I don't think this ever gets called except by things like nimbleparse.

Reply:
But yeah thinking about it, since we changed the signature of `add_rule` (and probably other things) incompatibly, it's probably worth changing this now.  To conform to a similar signature

Reply:
I think we either want one combined method or two separate ones (but not three). I'm neutral between one method or two.